browser/ui/views/sidebar/sidebar_container_view.cc

```diff
@@ -914,8 +902,12 @@ void SidebarContainerView::StartObservingContextualSidePanelEntry(
     return;
   }
 
-  for (const auto& entry : registry->entries()) {
-    StartObservingForEntry(entry.get());
+  // Adding observations for the side panel entries from tab not seen before.
+  if (!already_observed_tab_.count(tab->GetTabHandle())) {
+    already_observed_tab_.insert(tab->GetTabHandle());
```

Comment:
Same here. I think we don't need to have `already_observed_tab_`.
After we start observe a tab, that tab's registry could have new entry during that tab's lifetime.
We need to observe that newly added entry also.

Reply:
The issue with this one is that I would have to have sets of registration for certain tabs.

Reply:
Done. Ditto.